easybuild/tools/github.py

```diff
@@ -1591,6 +1591,24 @@ def update_pr(pr_id, paths, ecs, commit_msg=None):
     print_msg(msg, log=_log)
 
 
+def check_online_status():
+    """
+    Check whether we currently are online
+    Return True if online, else an URLError instance
+    """
+    # Try repeatedly and with different URLs to cater for flaky connections or servers
+    # Timeout and repeats set so total 1 minute
+    urls = [GITHUB_URL, GITHUB_API_URL]
+    for i in range(6):
+        try:
+            urlopen(urls[i % len(urls)], timeout=10)
+            result = True
+            break
+        except URLError as err:
```

Comment:
@Flamefire This won't even help with the 406 we were seeing, since that was a `HTTPError`, not a `URLError`...

Reply:
This is Python3 then I guess. Because you verified that URLError was caught "somewhere": https://github.com/easybuilders/easybuild-framework/pull/3192#issuecomment-583295750

Reply:
And a HTTPError is-a URLError: https://docs.python.org/3/library/urllib.error.html